src/store/validation/index.ts

```diff
@@ -0,0 +1,165 @@
+    const validate = getOrCreateBlockValidatorFor(blockTypeWithoutNameSpace)
+    const index = `block/${blockId}/`
+
+    Vue.set(state.validationStatuses, index, {
+      isValid: validate(block),
+      ajvErrors: validate.errors,
+    })
+
+    debugValidationStatus(state.validationStatuses[index], `validation status for ${index}`)
+    return state.validationStatuses[index]
+  },
+
+  async validate_flow({ state, commit }, { flow } : { flow: IFlow }): Promise<IValidationStatus> {
+    const validate = getOrCreateFlowValidator()
+    const index = `flow/${flow.uuid}/`
+    Vue.set(state.validationStatuses, index, {
+      isValid: validate(flow),
+      ajvErrors: validate.errors,
+    })
+
+    debugValidationStatus(state.validationStatuses[index], `flow validation status`)
+    return state.validationStatuses[index]
+  }
+}
+
+export const store: Module<IValidationState, IRootState> = {
+  namespaced: true,
+  state: stateFactory,
+  getters,
+  mutations,
+  actions,
+}
+
+export default store
+
+function getOrCreateBlockValidatorFor(blockType: string): ValidateFunction {
+  if (isEmpty(validators) || !validators.has(blockType)) {
+    const blockJsonSchemaFile = `I${blockType}Block.json`
+    // TODO: point to the right JSON once we consume the right flow-runner version, then delete tmp file
+    validators.set(blockType, createDefaultJsonSchemaValidatorFactoryFor(require(`../../../_tmp/${blockJsonSchemaFile}`)))
+  }
+  return validators.get(blockType)!
+}
+
+function getOrCreateFlowValidator(): ValidateFunction {
+  const validationType = 'flow'
+  if (isEmpty(validators) || !validators.has(validationType)) {
+    // TODO: point to the right JSON once we consume the right flow-runner version, then delete tmp file
+    validators.set(validationType, createDefaultJsonSchemaValidatorFactoryFor(require('../../../_tmp/flowSpecJsonSchema.json'), '#/definitions/IFlow'))
+  }
+  return validators.get(validationType)!
+}
+
+/**
+ * Create AJV Validator
+ * Usage :
+ * const validate = createDefaultJsonSchemaValidatorFactoryFor(require('./some-json-schema.json')
+ * const isValid = validate(myData)
+ * const error = validate.errors
+ *
+ * @param jsonSchema
+ * @param subSchema, Specify it if we want to pick a sub definition eg: pick `#/definitions/IFlow` under IContainer
+ */
+export function createDefaultJsonSchemaValidatorFactoryFor(jsonSchema: JSONSchema7, subSchema: string = ''): ValidateFunction {
+  if (subSchema === '') {
+    return ajv.compile(jsonSchema)
+  } else {
+    ajv.addSchema(jsonSchema)
+    const validate = ajv.getSchema(subSchema)
+    if (!validate) {
+      throw new Error(`Cannot find definition ${subSchema} in schema ${jsonSchema}`)
+    }
+    return validate as ValidateFunction
+  }
+}
+
+function debugValidationStatus(status: IValidationStatus, customMessage: string) {
+  if (status) {
+    console.debug(
+      customMessage,
```

Comment:
Can we try to create a pattern of using the first param to `console.*()` methods as the module path? This will help keep our log sane. Eg:
```
console.debug('store/validation', 'Uh oh! The thing happened again…')
```